biomeasure.py

```diff
@@ -1,29 +1,104 @@
 import pandas as pd
 import numpy
-
+import math
 from bioinput import Bioinput
+import sys
 
-class Biomeasure():
-    """
 
-    """
+class Biomeasure:
+    """ This is the main calling class
 
-    def __init__(self, threshold=0.9):
-        self.__hr_rawdata = pd.DataFrame(numpy.empty((10, 3))*numpy.nan, columns=['HeartRate', 'B/T', 'time'])
+    __init__ sets the __hr_rawdata
+
+    """
+    def __init__(self, threshold=0.8, thr_brady=50, thr_tachy=140):
         self.__threshold = threshold
-        input = Bioinput()
-        self.ecg_file = input.file
+        self.__thr_brady = thr_brady
+        self.__thr_tachy = thr_tachy
+        inputfile = Bioinput()
+        self.ecg_file = inputfile.file
+        self.__hr_rawdata = inputfile.read_input()
 
     def thresholdhr(self):
-        print(self.__threshold)
-        print(self.__hr_rawdata)
-        print(self.ecg_file)
+        """ Will return a list of thresholds, as well as the number of chunks and data_chunk size
+
+        :return: dataframe with column threshold_hr, and data chunk size and number of chunks for data
+        """
+
+        # self.__hr_rawdata['voltage'] = list(range(0, 100, 10))
+        # self.__hr_rawdata['time'] = list(range(0, 50, 5))
+        time_step = self.__hr_rawdata['time'][2] - self.__hr_rawdata['time'][1]
+        data_chunk = int(math.floor((5 / time_step)))
+        number_chunks = int(math.floor((len(self.__hr_rawdata) / data_chunk)))
+
+        threshold_hr = [None] * number_chunks
+
+        for j in range(0, number_chunks):
+            maxhr_data = max(self.__hr_rawdata['voltage'][(j * data_chunk):(j * data_chunk) + data_chunk])
+            threshold_hr[j] = self.__threshold * maxhr_data
+        threshold_hr = pd.DataFrame(numpy.array(threshold_hr), columns=['Threshold'])
+        return [threshold_hr, data_chunk, number_chunks]
```

Comment:
Again, variables are being returned instead of saving some of these as object attributes.  If they are metrics derived in the object, then they can/should be attributes.

Reply:
Now as an attribute as well